File: Code/Source/Estimator.Web/app/models/Section.ts
/// <reference path="../_ref.ts" />

module Estimator {
    "use strict";

    export interface ISection extends IModelBase {
        Name: string;
        Description?: string;

        TotalTasks: number;
        TotalHours: number;
        TotalWeeks: number;

        Estimates: IEstimate[];
    }

    export class Section extends ModelBase implements ISection {
        constructor() {
            super();
        }

        Name: string = '';
        Description: string;

        TotalTasks: number = 0;
        TotalHours: number = 0;
        TotalWeeks: number = 0;

        Estimates: IEstimate[] = [];
    }

}

File: Code/Source/Estimatorx.Core/Section.cs
using System;
using System.Collections.Generic;

namespace Estimatorx.Core
{
    /// <summary>
    /// A group of estimates for a section of the application being estimated.
    /// </summary>
    public class Section : ModelBase
    {
        public Section()
        {
            Tasks = new List<Task>();
        }

        /// <summary>
        /// Gets or sets the name of the section.
        /// </summary>
        /// <value>
        /// The name of the section.
        /// </value>
        public string Name { get; set; }
        
        /// <summary>
        /// Gets or sets the description for the section.
        /// </summary>
        /// <value>
        /// The description for the section.
        /// </value>
        public string Description { get; set; }

        
        /// <summary>
        /// Gets or sets the total number of tasks for this section.
        /// </summary>
        /// <value>
        /// The total number of tasks for this section.
        /// </value>
        public int TotalTasks { get; set; }
        
        /// <summary>
        /// Gets or sets the total number hours for this section.
        /// </summary>
        /// <value>
        /// The total number hours for this section.
        /// </value>
        public int TotalHours { get; set; }
        
        /// <summary>
        /// Gets or sets the total number of weeks for this section based on the number of hours per week for the project.
        /// </summary>
        /// <value>
        /// The total number of weeks for this section based on the number of hours per week for the project.
        /// </value>
        /// <seealso cref="P:Project.HoursPerWeek"/>
        public double TotalWeeks { get; set; }


        /// <summary>
        /// Gets or sets the estimates for this section.
        /// </summary>
        /// <value>
        /// The estimates for this section.
        /// </value>
        public List<Task> Tasks { get; set; }
    }
}